app/root.tsx

```diff
@@ -175,8 +188,35 @@ export default function App() {
   const managingAgency = searchParams.get(
     "managingAgency",
   ) as ManagingAgencyAcronym;
+  const commitmentsTotalMin = searchParams.get(
+    "commitmentsTotalMin",
+  ) as CommitmentsTotalMin;
+  const commitmentsTotalMax = searchParams.get(
+    "commitmentsTotalMax",
+  ) as CommitmentsTotalMax;
+  const {
+    commitmentsTotalMinInputValue,
+    commitmentsTotalMinSelectValue,
+    commitmentsTotalMaxInputValue,
+    commitmentsTotalMaxSelectValue,
+  } = handleCommitmentTotalsInputs(commitmentsTotalMin, commitmentsTotalMax);
+
+  const commitmentTotalInputsAreValid = checkCommitmentTotalInputsAreValid({
+    commitmentsTotalMinInputValue,
+    commitmentsTotalMinSelectValue,
+    commitmentsTotalMaxInputValue,
+    commitmentsTotalMaxSelectValue,
+  });
+
   const [attributeParams, setAttributeParams] = useState<AttributeParams>({
```

Comment:
Can you track `commitmentTotalInputsAreValid`, `commitmentsTotalMaxSelectValue`, and `commitmentsTotalMinSelectValue` separately from this `attributeParams` object? I was really picturing this object as responsible for tracking "things that end up as query parameters when you click search" and don't want it to become a catch-all for all of our state.

Reply:
Done - I also moved `commitmentsTotalMinInputValue` and `commitmentsTotalMaxInputValue`, as only `commitmentsTotalMin` and `commitmentsTotalMax` become searchParams